crates/sui-core/src/narwhal_manager/mod.rs

```diff
@@ -170,7 +174,10 @@ impl<TxValidator: TransactionValidator> NarwhalManager<TxValidator> {
 
                 // Drop storage
                 let path_clone = self.storage_base_path.clone();
-                spawn_monitored_task!(Self::remove_old_epoch_data(path_clone, epoch));
+                let handle = spawn_monitored_task!(Self::remove_old_epoch_data(path_clone, epoch));
+                let old_epoch_data_removal_handles =
+                    &*self.old_epoch_data_removal_handles.lock().await;
+                old_epoch_data_removal_handles.push(handle);
```

Comment:
I see that we are pushing the join handles in the FuturesUnordered , but it seems we are never accessing the structure to remove the handles. For example calling the `next` method to make progress on the future and remove the handle from there. I am thinking again what would be the best way to do this - especially after my recommendation - but unfortunately I can't see any other way unless we have a separate long running component/task that would be responsible for it.

So proposals:
Option 1. Yolo: we don't spend time for now, we just spin up the task, let it do it's thing and we hope it will be ok. 
Option 2. We add some extra safety we can just share an atomic counter to increment/decrement when we create/finish a job and throw an exception if we do have an excessive amount of concurrent tasks running - which is shouldn't under normal circumstances
Option 3. We spin up a separate component/task that it has as a sole purpose to receive the next epoch to delete - practically execute the `remove_old_epoch_data`. The communication can be done via a channel and the epochs can be sent there. In that way we also ensure that no multiple tasks will be running to delete data - but only one

Reply:
I might as well pull this out of the narwhal manager and have it be a standalone task that gets called for both Narwhal and Sui cleanup. cc @phoenix-o

Reply:
sorry for the late reply, we implemented somewhat similar PR for epoch databases: https://github.com/MystenLabs/sui/pull/7475
It has dedicated actor for cleanup as well
so alternatively, you can reuse our actor and plugin your parent directory

Reply:
We synced up before I started on this, and I made sure to tag you here with the intention of not duplicating work. What can we do to avoid this in the future? The PR linked above doesn't include things taken into consideration and required for NW cleanup, so unfortunately I won't be able to reuse that function. I wasn't able to mention these before merge either since I wasn't marked as a reviewer on your PR.

Reply:
no worries, it was kind of last minute call. We decided to ship it separately due to potential wave2 storage concerns. Initially plan was to ship that pruner later and reuse functionality of this PR to some extend. We can rethink it later and potentially merge pruners. Sorry for any confusion and inconvenience